### PR TITLE
Enabling Extended Session Distributed Tracing

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
I found that current implementation doesn't work well for extended session settings. 
This PR unlock the extended session with Distributed Tracing. 
Could you have a look, @cgillum ?

### CODE REVIEW

File: src/DurableTask.Core/TaskOrchestrationDispatcher.cs [hunk 1]
```diff
@@ -120,6 +120,16 @@ async Task OnProcessWorkItemSessionAsync(TaskOrchestrationWorkItem workItem)
                 }
 
                 var isExtendedSession = false;
+
+                CorrelationTraceClient.Propagate(
+                    () =>
+                    {                
+                        // Check if it is extended session.
+                        isExtendedSession = this.concurrentSessionLock.Acquire();
```
Comment (TsuyoshiUshio) [inline src/DurableTask.Core/TaskOrchestrationDispatcher.cs]: @cgillum Is there a better way to check if it is an extended session? 
I don't know why checking the lock availability becomes checking the extended session. It works, however, not cool.

File: src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs [hunk 2]
```diff
@@ -1095,6 +1105,37 @@ async Task AbandonAndReleaseSessionAsync(OrchestrationSession session)
...
+                            {
+                                var orchestrationDependencyTraceContext = traceContext.OrchestrationTraceContexts.Pop();
+                                CorrelationTraceClient.TrackDepencencyTelemetry(orchestrationDependencyTraceContext);
+                            }
+
+                            break;
+
+                        default:
+                            // When internal error happens, multiple message could come, however, it should not be prioritized.
+                            break;
```
Comment (ConnorMcMahon) [inline src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs]: I'm a bit confused why we want to break here. Is there a reason we don't want to go through the rest of the messages to check if they indicate the execution has finished?

File: src/DurableTask.Core/TaskOrchestrationDispatcher.cs (hunk 1 above)
Comment (ConnorMcMahon) [reply src/DurableTask.Core/TaskOrchestrationDispatcher.cs]: There might be... I will get back to you on that.

File: src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs (hunk 2 above)
Comment (ConnorMcMahon) [reply src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs]: Disregard this, the `break` is for the switch statement, NOT the for-loop. I think it is fine as is!

**Pull request merged**